src/solver/impls/petsc/petsc.cxx

```diff
@@ -359,26 +365,13 @@ int PetscSolver::init(int NOUT, BoutReal TIMESTEP) {
     ierr = PetscViewerDestroy(&fd);CHKERRQ(ierr);
   } else { // create Jacobian matrix
 
-    //PetscInt MXSUB = mesh->xend - mesh->xstart + 1;
-    //PetscInt MYSUB = mesh->yend - mesh->ystart + 1;
-
-    PetscInt nx = mesh->xend;//MXSUB;
-    PetscInt ny = mesh->yend;//MYSUB;
-
-    /* number of z points */
-    PetscInt nz  = mesh->LocalNz;
-
     /* number of degrees (variables) at each grid point */
     PetscInt dof = n3Dvars();
 
     /* Stencil width. Hardcoded to 2 until there is a public method to get mesh->MXG */
-    PetscInt n = local_N; //mesh->xend*mesh->yend*nz*dof; //<- that doesn't seem to work. Why is n3Dvars()*nz?
-    // PetscInt n = MXSUB*MYSUB*nz*dof;
     PetscInt sw = 2;
```

Comment:
Code comment above suggest the stencil width (`sw`) should be MXG, i.e. this should be `sw = mesh->xstart;`.